python-checks/src/main/java/org/sonar/python/checks/XPathCheck.java

```diff
@@ -50,20 +52,32 @@
     defaultValue = "" + DEFAULT_MESSAGE)
   public String message = DEFAULT_MESSAGE;
 
-  @Override
-  public String getXPathQuery() {
-    return xpathQuery;
-  }
+  private AstNodeXPathQuery<Object> query = null;
 
   @Override
-  public String getMessage() {
-    return message;
+  public void init() {
+    if (!Strings.isNullOrEmpty(xpathQuery)) {
+      try {
+        query = AstNodeXPathQuery.create(xpathQuery);
+      } catch (RuntimeException e) {
+        throw new IllegalStateException("Unable to initialize the XPath engine, perhaps because of an invalid query: " + xpathQuery, e);
+      }
+    }
   }
 
   @Override
   public void visitFile(@Nullable AstNode fileNode) {
-    if (fileNode != null) {
-      super.visitFile(fileNode);
+    if (query != null && fileNode != null) {
+      List<Object> objects = query.selectNodes(fileNode);
+
+      for (Object object : objects) {
+        if (object instanceof AstNode) {
+          AstNode astNode = (AstNode) object;
+          addLineIssue(message, astNode.getTokenLine());
+        } else if (object instanceof Boolean && (Boolean) object) {
+          addFileIssue(message);
+        }
```

Comment:
@ivandalbosco I basically copy/pasted the [implementation from sslr-squid-bridge](https://github.com/SonarSource/sslr-squid-bridge/blob/master/src/main/java/org/sonar/squidbridge/checks/AbstractXPathCheck.java) and changed how issues are created.
Can you create an XPath query which would give an AstNode or a Boolean?
If we were logging something here, it should not be more than debug level. Otherwise, it may create a lot of noise.